epoxy-adapter/src/main/java/com/airbnb/epoxy/AsyncListDifferWithPayload.java

```diff
@@ -0,0 +1,193 @@
+package com.airbnb.epoxy;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.util.DiffUtil;
+import android.support.v7.util.DiffUtil.ItemCallback;
+import android.support.v7.util.ListUpdateCallback;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+/**
+ * An adaptation of Google's {@link android.support.v7.recyclerview.extensions.AsyncListDiffer}
+ * that adds support for payloads in changes.
+ *
+ * Also adds support for canceling an in progress diff.
+ */
+class AsyncListDifferWithPayload<T> {
```

Comment:
Add a `@UiThread` annotation to the whole class?

Reply:
I actually want to figure out if we should support setting a new list from any thread. That could be nice when the models are built on a background thread. It would complicate this to synchronize things though. cc @gpeal

Reply:
I think the most important thing is that it happens off of the main thread rather than on a specific executor